core/lib/spree/core/importer/order.rb

```diff
@@ -169,117 +146,89 @@ def self.create_line_items_from_params(line_items, order)
-              payment.save!
-            rescue Exception => e
-              raise "Order import payments: #{e.message} #{p}"
-            end
+            payment = order.payments.build order: order
+            payment.amount = p[:amount].to_f
+            # Order API should be using state as that's the normal payment field. spree_wombat
+            # serializes payment state as status so imported orders should fall back to status field
+            payment.state = p[:state] || p[:status] || 'completed'
+            payment.payment_method = Spree::PaymentMethod.find_by_name!(p[:payment_method])
+            payment.source = create_source_payment_from_params(p[:source], payment) if p[:source]
+            payment.save!
           end
         end
 
         def self.create_source_payment_from_params(source_hash, payment)
-          begin
-            Spree::CreditCard.create(
-              month: source_hash[:month],
-              year: source_hash[:year],
-              cc_type: source_hash[:cc_type],
-              last_digits: source_hash[:last_digits],
-              name: source_hash[:name],
-              payment_method: payment.payment_method,
-              gateway_customer_profile_id: source_hash[:gateway_customer_profile_id],
-              gateway_payment_profile_id: source_hash[:gateway_payment_profile_id],
-              imported: true
-            )
-          rescue Exception => e
-            raise "Order import source payments: #{e.message} #{source_hash}"
-          end
+          Spree::CreditCard.create(
+            month: source_hash[:month],
+            year: source_hash[:year],
+            cc_type: source_hash[:cc_type],
+            last_digits: source_hash[:last_digits],
+            name: source_hash[:name],
+            payment_method: payment.payment_method,
+            gateway_customer_profile_id: source_hash[:gateway_customer_profile_id],
+            gateway_payment_profile_id: source_hash[:gateway_payment_profile_id],
+            imported: true
+          )
         end
 
         def self.ensure_variant_id_from_params(hash)
-          begin
-            sku = hash.delete(:sku)
-            unless hash[:variant_id].present?
-              hash[:variant_id] = Spree::Variant.active.find_by_sku!(sku).id
-            end
-            hash
-          rescue ActiveRecord::RecordNotFound => e
-            raise "Ensure order import variant: Variant w/SKU #{sku} not found."
-          rescue Exception => e
-            raise "Ensure order import variant: #{e.message} #{hash}"
+          sku = hash.delete(:sku)
+          unless hash[:variant_id].present?
+            hash[:variant_id] = Spree::Variant.active.find_by!(sku: sku).id
           end
+          hash
         end
 
         def self.ensure_country_id_from_params(address)
           return if address.nil? or address[:country_id].present? or address[:country].nil?
```

Comment:
Does this line actually work? Since we're using the `or` keyword instead of the `||` method, I'm thinking this is causing:

```ruby
(return if address.nil?) || address[:country_id].present? || address[:country].nil?
```